model/riscv_fdext_regs.sail

```diff
@@ -265,8 +265,8 @@ function rF_or_X_D(i) = {
   else if xlen >= 64
   then X(i)[63..0]
   else {
-    assert(i[0] == bitzero);
-    if i == zeros() then zeros() else X(i + 1) @ X(i)
+    assert((let Regidx(i_bits) = i in i_bits)[0] == bitzero);
```

Comment:
For `physaddr`/`virtaddr` we made a `physaddr_bits()` function.

```
function physaddr_bits(physaddr(paddr) : physaddr) -> physaddrbits = paddr

function virtaddr_bits(virtaddr(vaddr) : virtaddr) -> xlenbits = vaddr
```

Though I'm wondering it if makes sense to have a `bits_of` override function for all three cases?

We could do that in a new PR I guess. I think add a `regidx_bits` function though at least?

Reply:
`regidx_bits` is beyond my knowledge of how to do in `sail`, because it's type, `... regidx -> bits(N)` involves the `N` from the `bits(N)` within the definition of the `regidx` `newtype`.

Reply:
I think it would have the type `regidx -> {'n, 'in in {4, 5}. bits('n)}`, which you could read as `regidx` to some bitvector that has either length 4 or 5. It can't give a specific length because that's hidden inside regidx.

Reply:
Isn't `Regidx` always 5 bits currently? Or are you thinking of a future definition?

Reply:
Hm.  I was thinking of the I-or-E case, I think.  Indeed, for either always-I or always-E, it's trivial and I've introduced `regidx_bits` and used it where appropriate (I think).